oak_functions/loader/src/server.rs

```diff
@@ -43,6 +47,90 @@ type AbiPointerOffset = u32;
 /// Wasm would use a different value.
 const ABI_USIZE: ValueType = ValueType::I32;
 
+/// Specific header used for padding the response to match the expected response bytes
+const POLICY_PADDING_HEADER: &str = "oak-policy-padding";
+
+/// Minimum size of constant response bytes. It is large enough to fit an error response, in case
+/// the policy is failed.
+const MIN_RESPONSE_SIZE: u32 = 90;
+
+/// A policy describing limits on the size of the response and response processing time to avoid
+/// side-channel leaks.
+#[derive(Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Policy {
+    /// A fixed size for responses returned by the trusted runtime.
+    ///
+    /// If the response computed by the Wasm module is smaller than this amount, it is padded with
+    /// additional data before encryption in order to make the payload size exactly this size. If
+    /// the response is larger than this amount, the trusted runtime discards the response and
+    /// instead sends a response of exactly this size to the client, containing an error message
+    /// indicating the failure.
+    pub constant_response_size_bytes: u32,
+    /// A fixed response time.
+    ///
+    /// Similar to the previous one, but controls the amount of time the function is allowed to run
+    /// for. If the function finishes before this time, the response is not sent back until the
+    /// time is elapsed. If the function does not finish within this deadline, the trusted runtime
+    /// sends a response to the client containing an error message indicating the failure. The size
+    /// of this response is equal to the size specified by the previous parameter.
+    #[serde(with = "humantime_serde")]
+    pub constant_processing_time: Duration,
+}
+
+impl Policy {
+    pub fn validate(&self) -> anyhow::Result<()> {
+        match self.constant_response_size_bytes >= MIN_RESPONSE_SIZE {
+            true => Ok(()),
+            false => anyhow::bail!("Response size is too small"),
+        }
+    }
+
+    fn get_error_response(&self, reason: &str) -> anyhow::Result<Response<Body>> {
```

Comment:
AFAICT this is actually creating the error response, right (not really getting it from the policy), so maybe rename to make_ or create_ ? Also add some documentation?